python/tvm/relay/ty.py

```diff
@@ -93,6 +93,35 @@ class TypeConstraint(Type):
     pass
 
 
+@register_relay_node
+class TypeRelation(Type):
```

Comment:
Should TypeRelation actually inherit from TypeConstraint?

Reply:
yes, eventually we believe there to be more of these.